### ISSUE TITLE
`;` could be part of `jsx` node

### ISSUE THREAD

**Issue by JounQin** [mdx-js/eslint-mdx#183]
`<div>hi</div>;` is a valid jsx node, but `eslint-mdx` parses it incorrectly like `<div>hi</div>` what means `;` in jsx node are ignored.

**Comment by JounQin** [comment]
It seems `;` will be wrapped into a parent jsx too, so it makes the `;` symbol not to be same as a js expression. What means the following are different:

```js
export default <div></div>;

export default (
  <Wrapper>
    <div></div>;
  </Wrapper>
)
```

Does it mean we should always disable ESLint `semi` rule?

![image](https://user-images.githubusercontent.com/8336744/80191935-030ade80-8649-11ea-8a44-f18b3d55f53d.png)

@wooorm @johno @ChristopherBiscardi

**Comment by wooorm** [comment]
The semicolon is not part of JSX: https://facebook.github.io/jsx/
It’s part of JS with embedded JSX, because you can do `a = 1;`, you can also do `a = <>1</>;`.
So, for MDX, the semicolon shouldn’t be handled. Similar to how a plus, equals to, or other random characters wouldn’t be: `<>1</> + <>1</> = 2`

**Comment by JounQin** [comment]
Yes, so I think we need a new `mdx/semi` rule to replace the built-in `semi` rule which should work just same as the built-in one for plain js expression like `import/export` statements but reports warning if it's at the end of an unexported jsx expression.

```js
import Foo from 'foo'; // pass

<div></div>; // error

export const Bar = <div></div>; // pass
```

**Comment by wooorm** [comment]
Is it needed to fail on them? They’d appear as text in a post, and could be used:
```
The semicolon falls between <This /> and <That />; its strength is equal to that of the colon
```

**Comment by JounQin** [comment]
Sounds reasonable. So we should just care about semicolons in normal js expression without jsx?

**Comment by wooorm** [comment]
Yeah, I think so! JavaScript cn be in imports, exports, and expressions, semicolons should be checked there. But in other places (markdown or jsx), I don't think they should!

**Comment by JounQin** [comment]
Parsing is relied on `remark-mdx` totally now in v2.
